subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationExecutor.java

```diff
@@ -33,6 +33,39 @@
  */
 @ThreadSafe
 public interface BuildOperationExecutor extends BuildOperationRunner {
+    /**
+     * Runs the given build operation synchronously. Invokes the given operation from the current thread.
+     *
+     * <p>Rethrows any exception thrown by the action.
+     * Runtime exceptions are rethrown as is.
+     * Checked exceptions are wrapped in {@link BuildOperationInvocationException}.</p>
+     */
+    void run(RunnableBuildOperation buildOperation);
+
+    /**
+     * Calls the given build operation synchronously. Invokes the given operation from the current thread.
+     * Returns the result.
+     *
+     * <p>Rethrows any exception thrown by the action.
+     * Runtime exceptions are rethrown as is.
+     * Checked exceptions are wrapped in {@link BuildOperationInvocationException}.</p>
+     */
+    <T> T call(CallableBuildOperation<T> buildOperation);
```

Comment:
❓Isn't this a redeclaration from `BuildOperationRunner`?

Reply:
More or less. It adds a bunch of things about threads, basically.